internal/plugin.go

```diff
@@ -0,0 +1,227 @@
+	Severity          string   `json:"severity,omitempty" yaml:"severity,omitempty"`
+	Standards         []string `json:"standards,omitempty" yaml:"standards,omitempty"`
+}
+
+// Plugin is used to store the PolicyGenerator configuration and the methods to generate the
+// desired policies.
+type Plugin struct {
+	Metadata struct {
+		Name string `json:"name,omitempty" yaml:"name,omitempty"`
+	} `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+	PlacementBindingDefaults struct {
+		Name string `json:"name,omitempty" yaml:"name,omitempty"`
+	} `json:"placementBindingDefaults,omitempty" yaml:"placementBindingDefaults,omitempty"`
+	PolicyDefaults struct {
+		Categories        []string          `json:"categories,omitempty" yaml:"categories,omitempty"`
+		ComplianceType    string            `json:"complianceType,omitempty" yaml:"complianceType,omitempty"`
+		Controls          []string          `json:"controls,omitempty" yaml:"controls,omitempty"`
+		Namespace         string            `json:"namespace,omitempty" yaml:"namespace,omitempty"`
+		NamespaceSelector namespaceSelector `json:"namespaceSelector,omitempty" yaml:"namespaceSelector,omitempty"`
+		// This is named Placement so that eventually PlacementRules and Placements will be supported
+		Placement struct {
+			ClusterSelectors  map[string]string `json:"clusterSelectors,omitempty" yaml:"clusterSelectors,omitempty"`
+			PlacementRulePath string            `json:"placementRulePath,omitempty" yaml:"placementRulePath,omitempty"`
+		} `json:"placement,omitempty" yaml:"placement,omitempty"`
+		RemediationAction string   `json:"remediationAction,omitempty" yaml:"remediationAction,omitempty"`
+		Severity          string   `json:"severity,omitempty" yaml:"severity,omitempty"`
+		Standards         []string `json:"standards,omitempty" yaml:"standards,omitempty"`
+	} `json:"policyDefaults,omitempty" yaml:"policyDefaults,omitempty"`
+	Policies []policyConfig `json:"policies" yaml:"policies"`
+}
+
+// Config validates the input PolicyGenerator configuration, applies any missing defaults, and
+// configures the Policy object.
+func (p *Plugin) Config(config []byte) error {
+	err := yaml.Unmarshal(config, p)
+	if err != nil {
+		return fmt.Errorf("the PolicyGenerator configuration file is invalid: %w", err)
+	}
+	p.applyDefaults()
+
+	return p.assertValidConfig()
+}
+
+// applyDefaults applies any missing defaults under Policy.PlacementBindingDefaults and
+// Policy.PolicyDefaults. It then applies the defaults and user provided defaults on each
+// policy entry if they are not overridden by the user.
+func (p *Plugin) applyDefaults() {
+	if len(p.Policies) == 0 {
+		return
+	}
+
+	// Set defaults to the defaults that aren't overridden
+	if p.PlacementBindingDefaults.Name == "" && len(p.Policies) == 1 {
+		p.PlacementBindingDefaults.Name = "binding-" + p.Policies[0].Name
+	}
+
+	if p.PolicyDefaults.Categories == nil {
+		p.PolicyDefaults.Categories = []string{"CM Configuration Management"}
```

Comment:
Just think if we need to hardcode each default value here. Maybe define a default struct and use it here?

All rest LGTM

Reply:
/LGTM